frappe/tests/test_db_query.py

```diff
@@ -319,6 +319,10 @@ def test_between_filters(self):
 			filters={"creation": ["between", ["2016-07-06", "2016-07-07"]]}, fields=["name"]
 		)
 
+	def test_like_filter_for_datetime(self):
+		create_event(starts_on="2002-04-21 23:59:59")
+		self.assertTrue(frappe.get_all("Event", filters={"starts_on": ("like", "%2002-04-21%")}))
```

Comment:
let's not support like filtering on dates? it's very unintuitive. 

For starters, hardly anyone uses ISO standards for displaying dates, so users are naturally inclined to sort using display format and not the DB format.

Reply:
@sagarvora I doubt we can "parse" partial dates which are expected to be present in LIKE queries. Parsing will also go wrong with mm-dd and dd-mm :woozy_face: 

Lets just disallow this. Better to not half-arse this.

Reply:
> Parsing will also go wrong with mm-dd and dd-mm

Just clarifying: parsing will obviously only be done in user format, so this may not actually arise. Furthermore, partial dates like `20-01` are parse-able, but obviously not `20-0`. And in case we do parse this, it would be matched with equality, rather than like.